engine/src/render/shader/shader_bridge.cpp

```diff
@@ -0,0 +1,57 @@
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/render/shader/shader_bridge.hpp>
+
+using namespace cubos::engine;
+
+using cubos::core::memory::Stream;
+
+bool ShaderBridge::loadFromFile(Assets& assets, const AnyAsset& handle, Stream& stream)
+{
+    // Dump the shader code into a string.
+    std::string contents;
+    stream.readUntil(contents, nullptr);
+
+    // Read shader stage type
+    auto meta = assets.readMeta(handle);
+    auto stageTypeString = meta->get("stage");
+    cubos::core::gl::Stage stageType;
+    if (stageTypeString == "Vertex")
+    {
+        stageType = cubos::core::gl::Stage::Vertex;
+    }
+    else if (stageTypeString == "Geometry")
+    {
+        stageType = cubos::core::gl::Stage::Geometry;
+    }
+    else if (stageTypeString == "Pixel")
+    {
+        stageType = cubos::core::gl::Stage::Pixel;
+    }
+    else if (stageTypeString == "Compute")
+    {
+        stageType = cubos::core::gl::Stage::Compute;
+    }
+    else
+    {
+        CUBOS_ERROR("Shader assets must specify a stage type");
+        return false;
+    }
+
+    Shader shader(mRenderDevice.createShaderStage(stageType, contents.c_str()));
```

Comment:
Don't forget to handle the case where `createShaderStage` fails! In that case, it returns nullptr and you should return false.